test/types/mongodb.test-d.ts

```diff
@@ -11,3 +15,23 @@ expectDeprecated(Collection.prototype.remove);
 expectDeprecated(Collection.prototype.count);
 expectDeprecated(MongoError.create);
 expectDeprecated(AggregationCursor.prototype.geoNear);
+
+// test mapped cursor types
+const client = new MongoClient(process.env.MONGODB_URI || 'mongodb://localhost:27017');
+const coll = client.db('test').collection('test');
+const findCursor = coll.find();
+expectType<Promise<Document | null>>(findCursor.next());
```

Comment:
You can use `await` at the top level here, it would just cut down on the boilerplate `Promise<...>` wrapping.